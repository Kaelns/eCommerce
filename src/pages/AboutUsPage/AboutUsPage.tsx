import { Box, Icon, List, Stack, Card, CardMedia, CardContent, Link } from '@mui/material';
import { Title } from '@/components/typography/Title/Title';
import { TextBold } from '@/components/typography/TextBold/TextBold';
import { ABOUT_US } from '@/pages/AboutUsPage/AboutUs.constant';
import schoolLogo from '@/assets/rs_school.svg';

export function AboutUsPage(): React.ReactNode {
  return (
    <>
      <Title>About Us Page</Title>
      <Box sx={{ whiteSpace: 'pre-line' }}>
        <TextBold sx={{ textAlign: 'center' }}>
          All roads lead to
          <Link href="https://rs.school/">
            <Icon sx={{ height: 40, width: 100, ml: 20 }}>
              <Box component="img" src={schoolLogo} alt="RS School logo" />
            </Icon>
          </Link>
        </TextBold>
        {ABOUT_US.introduction}
        <List component={Stack} direction={{ xs: 'column', md: 'row' }}>
          {ABOUT_US.members.map((member) => (
            <Card key={member.lastName}>
              <CardMedia>
                <Box component="img" src={member.photo} alt={member.lastName} style={{ width: 200, height: 200 }} />
              </CardMedia>
              <CardContent>
                <Title variant="h6">{`${member.firstName} ${member.lastName}`}</Title>
                <TextBold> Github: </TextBold>
                <Link href={member.github} underline="hover">
                  {member.github}
                </Link>
                <TextBold> Role: </TextBold>
                {member.roles}
                <TextBold> Significant contributions: </TextBold>
                {member.contributions}
                <TextBold> BIO: </TextBold>
                {member.bio}
              </CardContent>
            </Card>
          ))}
        </List>
      </Box>
    </>
  );
}
