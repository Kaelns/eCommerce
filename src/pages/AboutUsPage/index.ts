export { AboutUsPage } from '@/pages/AboutUsPage/AboutUsPage';
