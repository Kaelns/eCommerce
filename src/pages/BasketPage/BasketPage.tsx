import { Box, Typography } from '@mui/material';
import { Title } from '@/components/typography/Title/Title';
import { useFetch } from '@/hooks/useFetch/useFetch';
import { fetchBasket } from '@/services/helpers/fetchBasket/fetchBasket';
import { LoadingFetch } from '@/components/LoadingFetch/LoadingFetch';
import { PageSkeleton } from '@/components/PageSkeleton/PageSkeleton';
import { ProductBasket } from '@/pages/BasketPage/components/ProductBasket/ProductBasket';
import { EMPTY_DATA_PRODUCTS } from '@/services/data/productsResponce/productsResponce.constants';

import styles from './BasketPage.module.scss';

export function BasketPage(): React.ReactNode {
  const { data, isLoading, error } = useFetch(fetchBasket);
  const { products, amount } = data ?? EMPTY_DATA_PRODUCTS;

  return (
    <LoadingFetch error={error} isLoading={isLoading} Skeleton={PageSkeleton} className={styles.productsContainer}>
      {products.length ? (
        <>
          <Box className={styles.productsHeader}>
            <Box>
              <Title className={styles.basket}>Product Basket</Title>
              <Typography>{amount} products</Typography>
            </Box>
          </Box>
          <Box>
            {products.map((product) => (
              <ProductBasket key={product.id} product={product} />
            ))}
          </Box>
        </>
      ) : (
        <Title>There is no products</Title>
      )}
    </LoadingFetch>
  );
}
