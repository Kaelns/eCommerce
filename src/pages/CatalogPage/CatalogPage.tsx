import { Box } from '@mui/material';
import { Title } from '@/components/Title/Title';
import { Breadcrumb } from '@/components/Breadcrumb/Breadcrumb';
import { arrayCategories } from '@/pages/CatalogPage/mock';

import styles from './CatalogPage.module.scss';

export function CatalogPage(): React.ReactNode {
  return (
    <Box className={styles.catalogContainer}>
      <Breadcrumb category={arrayCategories[10]} />
      <Title className={styles.catalog}>Catalog Page</Title>;
    </Box>
  );
}
