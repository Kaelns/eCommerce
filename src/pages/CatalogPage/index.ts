export { CatalogPage } from '@/pages/CatalogPage/CatalogPage';
export { ProductCard } from '@/pages/CatalogPage/components/ProductCard';
