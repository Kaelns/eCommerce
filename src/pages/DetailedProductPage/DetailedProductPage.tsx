import CloseIcon from '@mui/icons-material/Close';
import { Box, IconButton, Modal, Typography, useMediaQuery, useTheme } from '@mui/material';
import { useCallback, useContext, useMemo, useState } from 'react';
import { Stack } from '@mui/system';
import { useNavigate, useParams } from 'react-router-dom';
import { LoadingFetch } from '@/components/LoadingFetch';
import { ImgCarousel } from '@/components/ImgCarousel';
import { ECommerceContext } from '@/context/ECommerceContext/ECommerceContext';
import { findInCategories } from '@/services/helpers/findInCategories';
import { ProductHead } from '@/pages/DetailedProductPage/components/ProductHead';
import { LANGUAGE } from '@/services/ECommerceInitApi.constants';
import { ICreateImagesStyles, createImagesMap } from '@/pages/DetailedProductPage/DetailedProductPage.helpers';
import { SxPropsNotArr, SxStyles } from '@/shared/types';
import { Paths } from '@/shared/constants';
import { useFetch } from '@/hooks/useFetch/useFetch';
import { sxMixins } from '@/features/MuiTheme/mixins';
import { convertToLightProduct } from '@/services/helpers/convertToLightProduct';
import { getProductByKeyApi } from '@/services/model/products/getProductByKeyApi';
import { PageSkeleton } from '@/components/skeleton/PageSkeleton';

const sxStyles: SxStyles = {
  modal: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  modalBody: {
    position: 'relative',
    width: '93%',
    height: { zero: '54%', tablet: '80%', laptop: '93%' },
    padding: 1,
    borderRadius: 1,
    bgcolor: 'common.background'
  },
  imgSmall: {
    height: 100,
    width: 130
  },
  imgMedium: {
    height: 300
  },
  imgLarge: {
    height: { zero: '40vh', tablet: '65vh', laptop: '85vh' }
  },
  imgContainer: {
    position: 'relative',
    height: 300,
    borderRadius: 1,
    cursor: 'pointer',
    overflow: 'hidden',
    ...sxMixins.mediaHover(
      {
        transform: 'scale(1.05)'
      },
      '> img'
    )
  },
  closeIcon: {
    position: 'absolute',
    width: 50,
    height: 50,
    top: 0,
    right: 0,
    zIndex: 100,
    transform: { zero: 'translate(25%, -25%)', tablet: 'translate(50%, -50%)' },
    bgcolor: 'primary.main',
    ...sxMixins.mediaHover({
      bgcolor: 'primary.light'
    })
  }
};

export function DetailedProductPage(): React.ReactNode {
  const { id: key } = useParams();
  const navigate = useNavigate();
  if (!key) {
    navigate(Paths.ERROR);
  }

  const { data, isLoading, error } = useFetch(getProductByKeyApi, key!);
  const { categories } = useContext(ECommerceContext);
  const productData = useMemo(() => convertToLightProduct(data), [data]);
  // TODO change on categoriesObj
  const categoriesNames = useMemo(
    () => findInCategories(categories, productData.categoriesIdArr, true).map((obj) => obj.name[LANGUAGE]),
    [categories, productData.categoriesIdArr]
  );

  const theme = useTheme();
  const isMatchesMedia = useMediaQuery(theme.breakpoints.down('tablet'));

  const [open, setOpen] = useState(false);
  const [imgNum, setImgNum] = useState(0);
  const handleClose = (): void => setOpen(false);
  const handleOpen = useCallback(
    (num: number) => (): void => {
      setImgNum(num);
      setOpen(true);
    },
    []
  );

  const createImages = useCallback(
    (containerSize: SxPropsNotArr, onClick?: (num: number) => () => void) => {
      const stylesObj: ICreateImagesStyles = {
        containerStyles: [sxStyles.imgContainer, containerSize as object]
      };
      return createImagesMap(productData, stylesObj, onClick);
    },
    [productData]
  );

  return (
    <LoadingFetch error={error} isLoading={isLoading} Skeleton={PageSkeleton}>
      <Stack gap={1.5} flexDirection={{ zero: 'column-reverse', tablet: 'column' }}>
        <Stack direction="row" justifyContent="space-between" gap={1.5}>
          <ImgCarousel width={{ zero: 1, tablet: '65%' }} customDots={createImages(sxStyles.imgSmall)}>
            {createImages(sxStyles.imgMedium, handleOpen)}
          </ImgCarousel>
          {!isMatchesMedia && <ProductHead productData={productData} categoriesNames={categoriesNames} />}
        </Stack>

        <Box>
          {isMatchesMedia && <ProductHead productData={productData} categoriesNames={categoriesNames} />}
          <Typography>
            <b> &emsp;Description: </b>
            {productData.description}
          </Typography>
        </Box>

        <Modal open={open} onClose={handleClose} sx={sxStyles.modal}>
          <Stack direction="row" alignItems="center" justifyContent="center" sx={sxStyles.modalBody}>
            <IconButton onClick={handleClose} sx={sxStyles.closeIcon}>
              <CloseIcon />
            </IconButton>
            <ImgCarousel arrows openModalImg={imgNum}>
              {createImages(sxStyles.imgLarge)}
            </ImgCarousel>
          </Stack>
        </Modal>
      </Stack>
    </LoadingFetch>
  );
}
