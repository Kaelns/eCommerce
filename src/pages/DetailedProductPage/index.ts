export { DetailedProductPage } from '@/pages/DetailedProductPage/DetailedProductPage';
