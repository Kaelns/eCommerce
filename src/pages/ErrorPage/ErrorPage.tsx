import styles from './ErrorPage.module.scss';

// interface IProps {}

export function ErrorPage(/* props: IProps */): JSX.Element {
  return <h2 className={styles.error}>Error Page</h2>;
}
