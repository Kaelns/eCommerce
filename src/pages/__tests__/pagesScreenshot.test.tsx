import { render } from '@testing-library/react';
import { AboutUsPage } from '@/pages/AboutUsPage/AboutUsPage';
import { BasketPage } from '@/pages/BasketPage/BasketPage';
import { CatalogPage } from '@/pages/CatalogPage/CatalogPage';
import { DetailedProductPage } from '@/pages/DetailedProductPage/DetailedProductPage';
import { ErrorPage } from '@/pages/ErrorPage/ErrorPage';
import { LoginPage } from '@/pages/LoginPage/LoginPage';
import { MainPage } from '@/pages/MainPage/MainPage';
import { UserPage } from '@/pages/UserPage/UserPage';
import * as useAuthContextHook from '@/context/AuthContext/useAuthContext';

const mockedUseNavigate = vi.fn();
const mockedUseLocation = vi.fn();
vi.mock('react-router-dom', async () => {
  const mod = await vi.importActual<typeof import('react-router-dom')>('react-router-dom');
  return {
    ...mod,
    useNavigate: () => mockedUseNavigate,
    useLocation: () => mockedUseLocation
  };
});

describe('Given MainPage component', () => {
  it('When rendered, should match snapshot of AboutUsPage', () => {
    const { asFragment } = render(<AboutUsPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('When rendered, should match snapshot of BasketPage', () => {
    const { asFragment } = render(<BasketPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('When rendered, should match snapshot of CatalogPage', () => {
    const { asFragment } = render(<CatalogPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('When rendered, should match snapshot of DetailedProductPage', () => {
    const { asFragment } = render(<DetailedProductPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('When rendered, should match snapshot of ErrorPage', () => {
    const { asFragment } = render(<ErrorPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('When rendered, should match snapshot of LoginPage', () => {
    const useMoviesSpy = vi.spyOn(useAuthContextHook, 'useAuthContext');

    useMoviesSpy.mockReturnValue({
      authUserToken: '',
      setAuthUserToken: vi.fn()
    });

    const { asFragment } = render(<LoginPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('When rendered, should match snapshot of MainPage', () => {
    const { asFragment } = render(<MainPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('When rendered, should match snapshot of UserPage', () => {
    const { asFragment } = render(<UserPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
