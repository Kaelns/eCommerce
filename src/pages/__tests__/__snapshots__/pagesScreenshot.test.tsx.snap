// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given MainPage component > When rendered, should match snapshot of AboutUsPage 1`] = `
<DocumentFragment>
  <h5
    class="MuiTypography-root MuiTypography-h5 _about-us_3b65ea css-bezmlo-MuiTypography-root"
  >
    About Us Page
  </h5>
</DocumentFragment>
`;

exports[`Given MainPage component > When rendered, should match snapshot of BasketPage 1`] = `
<DocumentFragment>
  <h5
    class="MuiTypography-root MuiTypography-h5 _basket_01fa68 css-bezmlo-MuiTypography-root"
  >
    Basket Page
  </h5>
</DocumentFragment>
`;

exports[`Given MainPage component > When rendered, should match snapshot of CatalogPage 1`] = `
<DocumentFragment>
  <h5
    class="MuiTypography-root MuiTypography-h5 _catalog_b2ca2b css-bezmlo-MuiTypography-root"
  >
    Catalog Page
  </h5>
</DocumentFragment>
`;

exports[`Given MainPage component > When rendered, should match snapshot of DetailedProductPage 1`] = `
<DocumentFragment>
  <h5
    class="MuiTypography-root MuiTypography-h5 _detailed-product_0feb98 css-bezmlo-MuiTypography-root"
  >
    Detailed Product Page
  </h5>
</DocumentFragment>
`;

exports[`Given MainPage component > When rendered, should match snapshot of ErrorPage 1`] = `
<DocumentFragment>
  <div
    class="_container_4d13f2 MuiBox-root css-0"
  >
    <img
      alt="404"
      class="_image_4d13f2 MuiBox-root css-0"
      src="not-found.png"
    />
    <div
      class="MuiButtonGroup-root MuiButtonGroup-contained css-1d9bq45-MuiButtonGroup-root"
      role="group"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary _button_4d13f2 MuiButtonGroup-firstButton css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <div
          class="_container_c09763 MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="ArrowBackIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 _text_c09763  css-ahj2mt-MuiTypography-root"
          >
            Go back
          </p>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary _button_4d13f2 MuiButtonGroup-lastButton css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <div
          class="_container_c09763 MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 _text_c09763  css-ahj2mt-MuiTypography-root"
          >
            Go main
          </p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="ArrowForwardIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Given MainPage component > When rendered, should match snapshot of LoginPage 1`] = `
<DocumentFragment>
  <div
    class="_loginContainer_951041 MuiBox-root css-0"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
    >
      Login page
    </h5>
    <form
      class="_form_636167 MuiBox-root css-0"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
      >
        Enter your email
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          name="email"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root Mui-error css-1d1r5q-MuiFormHelperText-root"
      />
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
      >
        Enter your password
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          name="password"
          required=""
          type="password"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root Mui-error css-1d1r5q-MuiFormHelperText-root"
      />
      <div
        class="_btnContainer_636167 MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          Login
        </button>
        <div
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1puturl-MuiDivider-root"
          role="separator"
        >
          <span
            class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-13htn8f-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
              >
                Or
              </span>
            </div>
          </span>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Register
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Given MainPage component > When rendered, should match snapshot of MainPage 1`] = `
<DocumentFragment>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
  >
    Main page
  </h5>
  <nav
    class="MuiBox-root css-0"
  >
    <div
      class="MuiTabs-root MuiTabs-vertical css-1g4ac1o-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-right: 0px;"
      >
        <div
          aria-orientation="vertical"
          class="MuiTabs-flexContainer MuiTabs-flexContainerVertical css-lfwcke-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            Main
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            User
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Error
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Basket
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Catalog
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            About Us
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Register
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Detailed Product
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-107nzro-MuiTabs-indicator"
          style="top: 0px; height: 0px;"
        />
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Given MainPage component > When rendered, should match snapshot of UserPage 1`] = `
<DocumentFragment>
  <h5
    class="MuiTypography-root MuiTypography-h5 _user_11f9a9 css-bezmlo-MuiTypography-root"
  >
    User Page
  </h5>
</DocumentFragment>
`;
